text/0000-lifecycle-experimental-api-mode.md

```diff
@@ -0,0 +1,122 @@
+[apis.platform]
+  experimental = ["1.0-alpha1"]
+
+[lifecycle]
+  version = "0.10.0"
+```
+`experimental`:
+contains an array of experimental API versions
+the APIs will correspond to specs that may not be released and in branches that are still being worked on.
+the API version must be of the format: `<major>.<minor>-<alphanumeric>`, i.e. `1.0-alpha1`.
+
+Since experimental versions are defined with a different format, they will NOT automatically get upgraded to a `supported` release when the experimental API has finalized.
+
+## Lifecycle Labels
+This will extend the builder or lifecycle image with the above descriptor file will extend the label `io.buildpacks.lifecycle.apis` with the value:
+```json
+{
+  "buildpack": {
+    "experimental": ["0.6-rc1", "1.0-alpha1"],
+  },
+  "platform": {
+    "experimental": ["1.0-alpha1"],
+  }
+}
+```
+
+## API Types
+### Experimental APIs
+API versions aren't limited to those defined in a spec release, but can refer to upcoming spec work happening in a branch.
+
+New `CNB_PLATFORM_EXPERIMENTAL_MODE` and `CNB_BUILDPACK_EXPERIMENTAL` mode environment variables will control experimental mode with:
+allowed values: `warn`, `error`, `silent`
+default value: `warn`
+
+**When** the `CNB_PLATFROM_API` environment variable is set to an API version in the deprecated platform API, the lifecycle shall:
+ - **If** `CNB_PLATFORM_EXPERIMENTAL_MODE` is unset, **Then** print a warning and continue
+ - **If** `CNB_PLATFORM_EXPERIMENTAL_MODE=warn`, **Then** print a warning and continue
+ - **If** `CNB_PLATFORM_EXPERIMENTAL_MODE=error`, **Then** fail
+ - **If** `CNB_PLATFORM_EXPERIMENTAL_MODE=silent`, **Then** continue w/o warning
+
+**When** the `api` field in a `buildpack.toml` file is set to an API version in the deprecated buildpack API range the lifecycle shall:
+ - **If** `CNB_BUILDPACK_EXPERIMENTAL_MODE` is unset, **Then** print a warning and continue
+ - **If** `CNB_BUILDPACK_EXPERIMENTAL_MODE=warn`, **Then** print a warning and continue
+ - **If** `CNB_BUILDPACK_EXPERIMENTAL_MODE=error`, **Then** fail
+ - **If** `CNB_BUILDPACK_EXPERIMENTAL_MODE=silent`, **Then** continue w/o warning
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The technical implementation will follow what's outlined in [RFC#49](https://github.com/buildpacks/rfcs/blob/main/text/0049-multi-api-lifecycle-descriptor.md).
+
+# Drawbacks
+[drawbacks]: #drawbacks
+This will add more complexity/overhead that most users will not use.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Do Nothing
+If we choose to do nothing, we run the risk of releasing less tested API/spec releases.
+
+### Not a Different Version Format
+The Experimental API version format could instead follow the same format as deprecated and supported: `<major>-<minor>`. By doing this, users could automatically get upgraded from an experimental to supported/deprecated API version which may be surprising to them.
+
+### Unstable Features
```

Comment:
While there are things I like about this alternative I worry about combinatorial explosion with the API version. Can every unstable feature be used with every support API, are the interfaces the same in each case? etc. etc.

Reply:
I do think it limits what is possible for experimental. There are definitely "safer" cross cutting features and some that aren't. I just listed it as an alternative b/c it's something we could consider.